src/main/java/htsjdk/variant/variantcontext/GenotypeNumLikelihoodsCache.java

```diff
@@ -0,0 +1,43 @@
+package htsjdk.variant.variantcontext;
+
+import java.util.HashMap;
+
+/**
+ * Hybrid dynamic cache for genotype likelihood counts
+ */
+
+public class GenotypeNumLikelihoodsCache {
+
+    private final static int N_ALLELES = 5;
+    private final static int PLOIDY = 10;
+
+    private final int[][] staticCache = new int[5][10];
+    private final HashMap<Integer, HashMap<Integer, Integer>> dynamicCache = new HashMap<>();
```

Comment:
I don't know if we want to change this cache to do the calculations itself if missing.  I.e. instead of get/put, we could just have get which would handle the caching internally and always return an answer.  it might make things a bit cleaner.

Reply:
I like that idea.  I'll make that change